### ISSUE TITLE
Implement inherited-payload ladder (P2-P4) and run the #848 experiment

### ISSUE THREAD

**Issue by csmangum** [Dooders/AgentFarm#904]
Implementation + experiment tracker for richer inherited payloads (#848), following the first-principles design.

## Source context
- Design: `docs/design/inherited_payload_design.md`
- Devlog origin: `docs/devlog/2026-04-23-evolving-hyperparameter-genomes-foraging-learning-agents.md`
- A/B harness: `scripts/run_inheritance_mode_ab.py`, `scripts/compare_inheritance_arms.py`
- Protocol: `docs/experiments/intrinsic_evolution/inheritance_mode_ab.md`
- Early-life scoring: `scripts/analyze_early_life_fitness.py`

## Goal
Build the payload variants beyond the existing P0 (Baldwinian) / P1 (Lamarckian) and run the comparative experiment to answer #848: when does richer inheritance help vs. overfit to local ecology?

## Variant ladder (from the design doc)
- **P0** chromosome only (baseline, done)
- **P1** weights only (= current Lamarckian, done; already a fitness null)
- **P2** weights + plasticity damping (lower child LR/ε; carry step_count)
- **P3** weights + optimizer state + bounded replay slice (continuation machinery)
- **P4** gated/blended transfer (fitness/confidence gate; `theta_child = a*theta_parent + (1-a)*theta_init`)

## Dependencies (prereqs)
- [x] #901 — expose/load full module state (optimizer, replay slice, plasticity)
- [x] #902 — bounded, deterministic replay-buffer transfer
- [x] #903 — add P2-P4 `inheritance_mode` values + reproduction wiring

## Implementation checklist
- [x] P2 plasticity damping in the reproduction/warm-start path
- [x] P3 optimizer + replay-slice transfer
- [x] P4 gating + blended (soft) warm-start with tunable `alpha`
- [x] Per-mode warm-start telemetry (coverage, gate hit-rate, blend alpha)
- [x] Unit/integration tests for each mode

## Experiment checklist
- [x] **Precondition gate first:** measure the transferable-signal budget (greedy probe-states; end-of-life vs init policy quality). If ~null in the chosen regime, stop and record the negative result.
- [x] Run in a learning-positive regime (longer horizon, smaller population), not the default config where within-life learning is ~null (#878).
- [ ] Matched multi-seed cohorts (6-seed x 3-profile) across all arms.
- [ ] Cross-ecology transfer test (shifted resource density / regeneration / crowding) as the overfit discriminator.
- [ ] Aggregate with the standard robustness gate: paired (profile x seed) delta, 95% CI excludes zero AND sign agreement >= 75%.

## Trade-off readouts
- Adaptation speed: steps-to-threshold; net early RL reward at ages N in {10,25,50}.
- Stability: startup death rate, oscillation amplitude, lineage churn.
- Generalization: performance retention in held-out ecology.

## Decision rule
Escalate P1->P2->P3->P4 cheap-first; keep a richer payload only if it yields a robust early-life **net-RL-reward** gain without degrading cross-ecology generalization or stability.

## Deliverable
A devlog write-up (mirroring the 05-21 / 06-04 posts) reporting per-variant distributions, the cross-ecology result, and a clear verdict on when (if ever) richer inheritance beats Baldwinian.

Closes the implementation/experiment portion of #848.

**Comment by csmangum** [comment]
## Precondition gate: complete ✅

The transferable-signal precondition gate is done. Full write-up: [The transferable-signal gate: do learned policies beat their own init?](https://dooders.github.io/AgentFarm/research/devlog/2026-06-20-transferable-signal-budget/)

### Verdict: **PASS** (proceed to P2–P4 A/B)

In a learning-positive regime (8 agents, 3000 steps, reproduction disabled, 6 seeds × 3 STABLE profiles), end-of-life policies robustly beat freshly-initialized ones on a survival-decoupled decision metric — not on full-episode reward.

| Profile | Mean Δ reward@age10 | 95% CI | Sign agr. | Robust |
| --- | --- | --- | --- | --- |
| conservative | +15.6 | [12.3, 18.9] | 1.00 | yes |
| balanced | +17.8 | [14.2, 21.3] | 1.00 | yes |
| buffered | +18.1 | [15.5, 20.6] | 1.00 | yes |

Tier 1 (probe-state drift) also confirms policies genuinely moved: mean argmax disagreement ~0.83 (threshold 0.05).

### Important baseline correction

An early cut compared EOL vs init using **pure-greedy argmax** and reported a huge +566–616 budget. That was an artifact: greedy on an untrained net is a near-degenerate fixed-action controller that dies, so the gap was almost entirely survival, not decision quality. Re-running with the **real action-weighted policy** (`softmax(Q) × action_weights` — i.e. the Baldwinian P0 baseline) collapses the survival gap to zero and leaves a modest but real foraging edge (~+0.16 reward/step, ~+15–30 net reward by early ages). The gate keys off `reward_age_10_budget` for this reason.

### Reproduce

```bash
PYTHONHASHSEED=0 python scripts/measure_transferable_signal.py \
  --output-dir experiments/transferable_signal_v2
```

Runner: `scripts/measure_transferable_signal.py` · Plots: `scripts/plot_transferable_signal.py` · Tests: `tests/scripts/test_measure_transferable_signal.py`

### Implications for the remaining experiment checklist

- **Size the A/B for a small effect** (~3% per-step foraging gain), not the inflated +600 from the broken greedy baseline.
- **Use fair baselines** — each P2–P4 arm must beat Baldwinian P0 (weighted policy), not a strawman.
- **Grade on early-age net RL reward** at ages {10, 25, 50}, not full-episode reward or survival.
- **Cross-ecology probe** (`--cross-eval-profile`) is wired but not yet run as part of the main ladder A/B.

The "stop and record negative result" branch was not taken; there is measurable (if modest) transferable decision-quality signal worth testing whether P2–P4 can capture.

**Comment by csmangum** [comment]
## Potential future look at the ladder

The 07-08 / 07-09 experiments settled the question for **this** warm-start-on-reproduction design: none of P1–P4 beat Baldwinian cold-start in either the saturated or the sparse (low-churn) learning-positive regime, and the dose-response (richer payload → bigger loss) says the mechanism is maladaptive, not under-tuned. If we ever revisit inheritance, it should be as new, separately-scoped work along the lines the devlogs left open:

- **Blend-alpha sweep toward 0 (P4).** At blend 0.5 the child gets half the parent policy; sweeping alpha → 0 interpolates back to P0. If even a small blend hurts in `buffered`, the imposition itself is the problem regardless of dose — a cheap, decisive experiment on the existing harness.
- **Per-offspring warm-start telemetry.** Coverage is ~100%, so arm-level comparisons can't separate "warm-started" from "born via the treatment path." Per-offspring applied/skipped logging would let us contrast warm-started vs. cold offspring *within* a run (also relevant to the `incompatible_state` skip issue).
- **Different transfer designs.** The barrier established here is specific to transplanting the parent's operating point at birth. Untested alternatives: distilled priors (train the child briefly against parent behavior rather than copying weights), delayed fitness-gated transfer (only after the child proves viable), and partial-layer warm-start (transfer early feature layers, cold-start the head).

Reference: [inheritance ladder A/B](https://dooders.github.io/AgentFarm/research/devlog/2026-07-08-inheritance-ladder-warm-start-clamps-offspring/) and the [low-churn follow-up](https://dooders.github.io/AgentFarm/research/devlog/2026-07-09-lowchurn-inheritance-still-loses/).
